### PR TITLE
Added script that will attach an entity to a vive sensor

### PR DESCRIPTION
**Pull request by hyperlogic**
This tablet script/app can be used to attach a model entity to a vive sensor.

### CODE REVIEW

**Comment by hyperlogic** [comment]
QA Test: (vive pucks required)
  * Run puck-attach.js
  * In the Puck Number field enter the number of the puck.  If only one puck is attached this number is usually 5.
  * Press the Create Attachment button to spawn a new entity (specified by the Model URL field) at the pucks location / rotation.
  * Test the position and rotation offset fields.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/931/HighFidelity-Beta-PR10763-b78e2781020c44737c3e406de83b5675361433e9.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/931/HighFidelity-Beta-PR10763-b78e2781020c44737c3e406de83b5675361433e9.dmg

File: scripts/developer/tests/puck-attach.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    if (attachedEntity) {
+        Script.update.disconnect(update);
+        Entities.deleteEntity(attachedEntity);
+        attachedEntity = undefined;
+    }
+    attachedObj = undefined;
+}
+
+function pad(num, size) {
+    var s = "000000000" + num;
```
Comment (ZappoMan) [inline scripts/developer/tests/puck-attach.js]: coding standard says what?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1008/HighFidelity-Beta-PR10763-3184d4e9d13ae105e531fffff9a4e61959cb48be.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1008/HighFidelity-Beta-PR10763-3184d4e9d13ae105e531fffff9a4e61959cb48be.exe

File: scripts/developer/tests/puck-attach.js [hunk 2]
```diff
@@ -13,7 +13,8 @@ Script.include("/~/system/libraries/Xform.js");
 (function() { // BEGIN LOCAL_SCOPE
 
 var TABLET_BUTTON_NAME = "PUCKTACH";
-var HTML_URL = "https://s3.amazonaws.com/hifi-public/tony/html/puck-attach.html";
+// var HTML_URL = "https://s3.amazonaws.com/hifi-public/tony/html/puck-attach.html";
+var HTML_URL = "file:///C:/msys64/home/anthony/code/hifi/examples/html/puck-attach.html";
```
Comment (ZappoMan) [inline scripts/developer/tests/puck-attach.js]: oops?

Comment (hyperlogic) [inline scripts/developer/tests/puck-attach.js]: good find

Comment (anshuman64) [inline scripts/developer/tests/puck-attach.js]: lol

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1014/HighFidelity-Beta-PR10763-e531468ac7dacbe5b99e5292141a8f0158f13e8a.dmg

**Comment by hyperlogic** [comment]
No cpp code has changed the windows build should not have failed

test this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1029/HighFidelity-Beta-PR10763-e531468ac7dacbe5b99e5292141a8f0158f13e8a.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1029/HighFidelity-Beta-PR10763-e531468ac7dacbe5b99e5292141a8f0158f13e8a.dmg

File: scripts/developer/tests/puck-attach.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
+//
+//  Created by Anthony J. Thibault on 2017/06/20
+//  Copyright 2017 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
```
Comment (PhilipRosedale) [inline scripts/developer/tests/puck-attach.js]: Explain how to use this script up here?

File: scripts/developer/tests/puck-attach.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        if (!shown) {
+            // hook up to event bridge
+            tablet.webEventReceived.connect(onWebEventReceived);
+
+            Script.setTimeout(function () {
+                // send available tracked objects to the html running in the tablet.
+                var availableTrackedObjects = getAvailableTrackedObjects();
+                tablet.emitScriptEvent(JSON.stringify(availableTrackedObjects));
+
+                print("PUCK-ATTACH: availableTrackedObjects = " + JSON.stringify(availableTrackedObjects));
```
Comment (PhilipRosedale) [inline scripts/developer/tests/puck-attach.js]: Quiet debug prints

File: scripts/developer/tests/puck-attach.js [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    var localPos = {x: Number(obj.posx), y: Number(obj.posy), z: Number(obj.posz)};
+    var localRot = Quat.fromVec3Degrees({x: Number(obj.rotx), y: Number(obj.roty), z: Number(obj.rotz)});
+    attachedObj.localXform = new Xform(localRot, localPos);
+    var key = indexToTrackedObjectName(Number(attachedObj.puckno));
+    attachedObj.key = key;
+
+    print("PUCK-ATTACH: attachedObj = " + JSON.stringify(attachedObj));
+
+    Script.update.connect(update);
+    update(0.001);
```
Comment (PhilipRosedale) [inline scripts/developer/tests/puck-attach.js]: Magic number... 1000fps?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1081/HighFidelity-Beta-PR10763-aabbcfd23dafecd99beb85a6c3ab3cf7a7e6b3dc.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1081/HighFidelity-Beta-PR10763-aabbcfd23dafecd99beb85a6c3ab3cf7a7e6b3dc.exe

**Pull request merged**